src/core/algorithms/cfd/cfd_verifier/cfd_stats_calculator.cpp

```diff
@@ -0,0 +1,119 @@
+#include "cfd_stats_calculator.h"
+
+#include "cfd/util/cfd_output_util.h"
+
+namespace algos::cfd_verifier {
+void CFDStatsCalculator::CreateSupportMask() {
+    support_mask_.resize(relation_->GetNumRows(), true);
+
+    for (size_t i = 0; i < relation_->GetNumRows(); i++) {
+        bool supports_cfd = true;
+
+        std::vector<int> row_values = relation_->GetRow(i);
+
+        for (auto expected_value : rule_.first) {
+            if (std::find(row_values.begin(), row_values.end(), expected_value) ==
+                        row_values.end() &&
+                expected_value > 0) {
+                supports_cfd = false;
+                break;
+            }
+        }
+
+        support_mask_[i] = supports_cfd;
+    }
+}
+
+void CFDStatsCalculator::MakeLhsToRowNums() {
+    for (size_t row_idx = 0; row_idx < relation_->GetNumRows(); ++row_idx) {
+        if (!support_mask_[row_idx]) {
+            continue;
+        }
+
+        cfd::Itemset lhs_values;
+        auto const& row_values = relation_->GetRow(row_idx);
+        for (int attr_idx : lhs_attrs_) {
+            lhs_values.push_back(row_values[attr_idx]);
+        }
+
+        lhs_to_row_nums_[lhs_values].push_back(row_idx);
```

Comment:
I believe that will be better since `lhs_values` is a vector we no longer need
```suggestion
        lhs_to_row_nums_[std::move(lhs_values)].push_back(row_idx);
```